### ISSUE TITLE
Parser mis-interprets KEY columns as MySQL KEY (INDEX) specification if unquoted and of type VARCHAR(n)

### ISSUE THREAD

**Issue by mircowidmer** [jOOQ/jOOQ#12712]
Hi

I just upgraded my project from 3.14.11 to 3.15.0 and encountered the following issue:

```
[ERROR] An exception occurred while parsing a DDL script: Token ')' expected: [5:19] ...Y KEY,
        fkView BIGINT not null,
        key VARCHAR(100) [*]not null,
        description VARCHAR(2000) not null,
        type VARCHAR(100) not null,
        cha.... Please report this error to https://github.com/jOOQ/jOOQ/issues/new
```

This happens while generating the Java classes for the following table (SQLite v 3.34.0):

```
CREATE TABLE Chart
(
	id INTEGER PRIMARY KEY,
	fkView BIGINT not null,
	key VARCHAR(100) not null,
	description VARCHAR(2000) not null,
	type VARCHAR(100) not null,
	chart VARCHAR(4000) not null,
	FOREIGN KEY (fkView) REFERENCES View (id) ON DELETE CASCADE,
	UNIQUE (fkView, key)
);
```

I couldn't find any reference to a breaking change here: https://www.jooq.org/notes

**Comment by lukaseder** [comment]
Thanks a lot for your report.

This is related to https://github.com/jOOQ/jOOQ/issues/7348. A similar fix was implemented with https://github.com/jOOQ/jOOQ/issues/7651, implementing a parser lookahead for some things that aren't MySQL `KEY` declarations, but the fix does not work for types like `VARCHAR(n)` yet.

For example, this works:

```sql
key INTEGER not null,
```

But so does this:

```sql
"key" VARCHAR(100) not null,
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.6 (https://github.com/jOOQ/jOOQ/issues/12717). Thanks again for your report!
